go/interpreter/lfvm/instructions_test.go

```diff
@@ -1997,6 +1939,69 @@ func TestInstructions_Sha3_WritesCorrectHashInStack(t *testing.T) {
 	}
 }
 
+func TestInstructions_opLog(t *testing.T) {
+
+	tests := map[string]struct {
+		offset, size  *uint256.Int
+		expectedError error
+		reduceGas     uint64
+	}{
+		"returns error if memory expansion fails": {
+			offset: uint256.NewInt(math.MaxUint64), size: uint256.NewInt(1),
+			expectedError: errOverflow,
+		},
+		"returns error if word gas cost is more than available gas": {
+			offset: uint256.NewInt(10), size: uint256.NewInt(128),
+			reduceGas:     1,
+			expectedError: errOutOfGas,
+		},
+		"calls emitLog with recipient": {
+			offset: uint256.NewInt(10), size: uint256.NewInt(128),
+		},
+	}
+	for name, test := range tests {
+		for n := 0; n < 4; n++ {
+			t.Run(fmt.Sprintf("%v/LOG%d", name, n), func(t *testing.T) {
+
+				runContext := tosca.NewMockRunContext(gomock.NewController(t))
+				ctxt := getEmptyContext()
+				for i := 0; i < n; i++ {
+					ctxt.stack.push(uint256.NewInt(0))
+				}
+				ctxt.stack.push(test.size)
+				ctxt.stack.push(test.offset)
+				ctxt.memory.store = make([]byte, 10)
```

Comment:
nit: this actually gets the memory in an invalid state; a memory with data of size 10 should never exist; maybe use `NewMemory` instead?

Reply:
Using `memory.set` now, 
`set` error is ignored, because if caught, it would shadow the logn error cases.